### PR TITLE
Staking V4: auction list api endpoint

### PR DESCRIPTION
**Pull request by bogdan-rosianu**
implemented the first version of the auction list API endpoint

### CODE REVIEW

File: factory/disabled/stakingDataProvider.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
+package disabled
+
+import (
+	"math/big"
+
+	"github.com/ElrondNetwork/elrond-go/state"
+)
+
+var emptyBI = big.NewInt(0)
```
Comment (mariusmihaic) [inline factory/disabled/stakingDataProvider.go:9]: This name seems a bit unfortunate, can we change it to `zero`?

**Review by mariusmihaic — COMMENTED** [review]
GJ by deleting duplicated mocks

File: api/groups/validatorGroup_test.go [hunk 2]
```diff
@@ -106,12 +114,76 @@ func TestValidatorStatistics_ReturnsSuccessfully(t *testing.T) {
 	assert.Equal(t, validatorStatistics.Result, mapToReturn)
 }
 
+func TestAuctionList_ErrorWhenFacadeFails(t *testing.T) {
+	t.Parallel()
+
+	errStr := "error in facade"
+
+	facade := mock.FacadeStub{
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply api/groups/validatorGroup_test.go:122]: done

File: api/groups/validatorGroup_test.go [hunk 3]
```diff
@@ -106,12 +114,76 @@ func TestValidatorStatistics_ReturnsSuccessfully(t *testing.T) {
...
+	}
+
+	validatorGroup, err := groups.NewValidatorGroup(&facade)
+	require.NoError(t, err)
+
+	ws := startWebServer(validatorGroup, "validator", getValidatorRoutesConfig())
+
+	req, _ := http.NewRequest("GET", "/validator/auction", nil)
+
+	resp := httptest.NewRecorder()
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply api/groups/validatorGroup_test.go:135]: done

File: api/groups/validatorGroup_test.go [hunk 4]
```diff
@@ -106,12 +114,76 @@ func TestValidatorStatistics_ReturnsSuccessfully(t *testing.T) {
...
+
+	auctionListToReturn := []*common.AuctionListValidatorAPIResponse{
+		{
+			Owner:   "owner",
+			NodeKey: "nodeKey",
+			TopUp:   "112233",
+		},
+	}
+
+	facade := mock.FacadeStub{
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply api/groups/validatorGroup_test.go:156]: done

File: api/groups/validatorGroup_test.go [hunk 5]
```diff
@@ -106,12 +114,76 @@ func TestValidatorStatistics_ReturnsSuccessfully(t *testing.T) {
...
+	}
+
+	validatorGroup, err := groups.NewValidatorGroup(&facade)
+	require.NoError(t, err)
+
+	ws := startWebServer(validatorGroup, "validator", getValidatorRoutesConfig())
+
+	req, _ := http.NewRequest("GET", "/validator/auction", nil)
+
+	resp := httptest.NewRecorder()
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply api/groups/validatorGroup_test.go:169]: done

**Review by gabi-vuls — DISMISSED** [review]
System test passed. 
@@                            Log scanner                                  @@
##                            auction-list-api-endpoint                    ##
================================================================================
- Known Warnings              10
- New Warnings                2
- Known Errors                0
- New Errors                  0
- Panics                      0
================================================================================

File: api/mock/facadeStub.go [hunk 6]
```diff
@@ -287,6 +288,11 @@ func (f *FacadeStub) ValidatorStatisticsApi() (map[string]*state.ValidatorApiRes
 	return f.ValidatorStatisticsHandler()
 }
 
+// AuctionListApi is the mock implementation of a handler's AuctionListApi method
+func (f *FacadeStub) AuctionListApi() ([]*common.AuctionListValidatorAPIResponse, error) {
+	return f.AuctionListHandler()
+}
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply api/mock/facadeStub.go:294]: done, for the entire file

File: common/dtos.go [hunk 7]
```diff
@@ -13,3 +13,10 @@ type TransactionsPoolAPIResponse struct {
 	SmartContractResults []string `json:"smartContractResults"`
 	Rewards              []string `json:"rewards"`
 }
+
+// AuctionListValidatorAPIResponse holds the data needed for an auction node validator for responding to API calls
+type AuctionListValidatorAPIResponse struct {
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply common/dtos.go:18]: no, I think the additional `Validator` that you want me to remove brings a good context to the naming, so I would leave it like this

File: epochStart/metachain/systemSCs.go [hunk 8]
```diff
@@ -343,9 +342,9 @@ func compareByXORWithRandomness(pubKey1, pubKey2, randomness []byte) bool {
 }
 
 func (s *systemSCProcessor) displayAuctionList(auctionList []state.ValidatorInfoHandler, numOfSelectedNodes uint32) {
-	if log.GetLevel() > logger.LogDebug {
-		return
-	}
+	//if log.GetLevel() > logger.LogDebug {
+	//	return
+	//}
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply epochStart/metachain/systemSCs.go:347]: right. thanks

File: epochStart/metachain/systemSCs.go [hunk 9]
```diff
@@ -375,7 +374,7 @@ func (s *systemSCProcessor) displayAuctionList(auctionList []state.ValidatorInfo
 	}
 
 	message := fmt.Sprintf("Auction list\n%s", table)
-	log.Debug(message)
+	log.Info(message)
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply epochStart/metachain/systemSCs.go:377]: done

File: facade/mock/nodeStub.go [hunk 10]
```diff
@@ -166,6 +167,11 @@ func (ns *NodeStub) ValidatorStatisticsApi() (map[string]*state.ValidatorApiResp
 	return ns.ValidatorStatisticsApiCalled()
 }
 
+// AuctionListApi -
+func (ns *NodeStub) AuctionListApi() ([]*common.AuctionListValidatorAPIResponse, error) {
+	return ns.AuctionListApiCalled()
+}
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply facade/mock/nodeStub.go:173]: changed for the entire file

File: testscommon/stakingcommon/stakingDataProviderStub.go [hunk 11]
```diff
@@ -0,0 +1,92 @@
+package stakingcommon
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply testscommon/stakingcommon/stakingDataProviderStub.go:1]: right. done

File: factory/disabled/stakingDataProvider.go (hunk 1 above)
Comment (bogdan-rosianu) [reply factory/disabled/stakingDataProvider.go:9]: renamed to `zeroBI`

File: api/mock/facadeStub.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (mariusmihaic) [reply api/mock/facadeStub.go]: :+1:

**Review by mariusmihaic — APPROVED** [review]

**Review by mariusmihaic — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**